File: sdk/dotnet-guest/templates/http-status.cs
using System.Collections.Generic;
using ServiceWorld;
using Lsf.Guest;
using Raw = ServiceWorld.wit.Imports.latent.http.IClientImports;
namespace ServiceWorld.wit.Exports.examples.httpStatus;

public class ApiExportsImpl : IApiExports
{
    public static Result<ushort, Raw.HttpError> Check(string url)
    {
        var response = Http.Send(new Raw.Request(
            Raw.Method.GET, url, new List<Raw.Header>(), null, null, null, 5000));
        return response.IsOk ? Result<ushort, Raw.HttpError>.Ok(response.AsOk.status)
            : Result<ushort, Raw.HttpError>.Err(response.AsErr);
    }
}

File: sdk/java-guest/templates/http-status.java
package dev.latent.app;
import dev.latent.generated.Bindings;
import dev.latent.guest.Option;
import dev.latent.guest.Result;
import java.util.List;

// lsf-example-begin: capsule
public final class Capsule implements Bindings.Exports {
    public Result<Integer, Bindings.LatentHttpClientHttpError> check(String url) {
        var request = new Bindings.LatentHttpClientRequest(Bindings.LatentHttpClientMethod.Get,
            url, List.of(), Option.none(), Option.none(), Option.none(), Option.none());
        var response = Bindings.LatentHttpClient.send(request);
        if (response.isError()) return Result.err(response.error());
        // No automatic retries, URL policy shortcuts, or secret logging.
        return Result.ok(response.value().status());
    }
}
// lsf-example-end: capsule
